src/evaluatorfrontend.cpp

```diff
@@ -100,41 +103,54 @@ void EvaluatorFrontend::RunRTEvaluator() {
 	auto endTime = std::chrono::steady_clock::now();
 	bool remainingStatesNotAdded = true;
 
+	boost::lockfree::queue<NetworkState *> stateQueue(1024);
+
+	if (!stateQueue.is_lock_free()) {
+		throw std::runtime_error("Queue is not lockfree");
+	}
+
+	std::thread threadObj([this, &stateQueue] {
+		while (!evaluator->IsFinished()) {
+			states.push_back(evaluator->GetNextNetworkState());
+			if (printStd)
+				std::cout << states.back().PrintCsvRow();
+			NetworkState *copy = new NetworkState(states.back());
+			if (!stateQueue.push(copy)) {
+				throw std::runtime_error("Push failed");
+			}
+		}
+	});
+
 	while (!mPlot->RunPlot()) {
-		if (EvaluatorFunc()) {
-			endTime = std::chrono::steady_clock::now();
-			auto elapsedTime = std::chrono::duration_cast<std::chrono::milliseconds>(
-					endTime - startTime);
-			auto updateTime = std::chrono::milliseconds(UpdateRate);
+		endTime = std::chrono::steady_clock::now();
+		auto elapsedTime = std::chrono::duration_cast<std::chrono::milliseconds>(
+				endTime - startTime);
+		auto updateTime = std::chrono::milliseconds(UpdateRate);
 
-			if (elapsedTime > updateTime) {
-				startTime = std::chrono::steady_clock::now();
+		if (elapsedTime > updateTime) {
+			startTime = std::chrono::steady_clock::now();
```

Comment:
This check will run until the program exits, that is not really the intention, since it only really matter when toPlot can expect an update (That is as long as the simulation is running)